staticd/static_vty.c

```diff
@@ -294,8 +294,16 @@ static int static_route_leak(struct vty *vty, const char *svrf,
 				 buf_gate_str, ifname);
 
 		dnode = yang_dnode_get(vty->candidate_config->dnode, ab_xpath);
-		if (!dnode)
+		if (!dnode) {
+			if (vty)
```

Comment:
1. There is no need to check for `vty` here, the function is always called from DEFUNs.
2. Are you sure we need this error at all? It looks like staticd never complained about the removal of the non-existing route.

Reply:
1. Good point, I'll address that
2. The use-case here is when a user doesn't specify a distance to an installed route which has a configured has a dist. Kinda like this:

```
w@kvm-gentoo ~/code $ vtysh

Hello, this is FRRouting (version 7.7-dev).
Copyright 1996-2005 Kunihiro Ishiguro, et al.

kvm-gentoo# c
kvm-gentoo(config)# ip route 2.2.2.2/24 1.1.1.5 10
kvm-gentoo(config)# do show ip route 2.2.2.2/24 
Routing entry for 2.2.2.0/24
  Known via "static", distance 10, metric 0
  Last update 00:00:08 ago
    1.1.1.5, via dummy0, weight 1

Routing entry for 2.2.2.0/24
  Known via "connected", distance 0, metric 0, best
  Last update 00:13:00 ago
  * directly connected, dummy1

kvm-gentoo(config)# no ip route 2.2.2.0/24 1.1.1.1
% Refusing to remove a non-existent route
```

Reply:
Got it, seems reasonable.